VoodooI2CHID/VoodooI2CMultitouchHIDEventDriver.cpp

```diff
@@ -23,12 +23,18 @@ OSDefineMetaClassAndStructors(VoodooI2CMultitouchHIDEventDriver, IOHIDEventServi
 
 AbsoluteTime last_multi_touch_event = 0;
 
-static int pow(int x, int y) {
-    int ret = 1;
-    while (y > 0) {
-        ret *= x;
-        y--;
+static int scientific_pow(UInt32 significand, UInt32 base, SInt32 exponent) {
```

Comment:
Can we not use any built-in to STL  for this function? 

I don't recall if cmath or the C++ equivalent has it.

Reply:
I couldn't find one that matches. We could use floats, but I'm not sure if using floating point arithmetics is safe in the kernel. I know it isn't in Linux.

IIRC the former method was also our own to avoid linking issues between OS versions. Not sure using MacKernelSDK will currently solve those.

Reply:
Ah okay. We lose precision and performance by using floats, let's not.